dhcpv6/ztpv6/parse_vendor_options.go

```diff
@@ -66,6 +69,27 @@ func ParseVendorData(packet dhcpv6.DHCPv6) (*VendorData, error) {
 			vd.Model = p[1]
 			vd.Serial = p[2]
 			return &vd, nil
+
+		// For Ciena the class identifier (opt 60) is written in the following format:
+		//    {vendor iana code}-{product}-{type}
+		// For Ciena the iana code is 1271
+		// The product type is a number that maps to a Ciena product
+		// The type is used to identified different subtype of the product.
+		// An example can be ‘1271-23422Z11-123’.
+		case strings.HasPrefix(d, strconv.Itoa(int(iana.EnterpriseIDCienaCorporation))):
+			v := strings.Split(d, "-")
+			if len(v) < 3 {
+				return nil, errVendorOptionMalformed
+			}
+			innerMessage, err := packet.GetInnerMessage()
```

Comment:
The only thing I am not convinced about is how we get the DUID.

In here we get the inner message but we don't do that for the opt16 and opt17 above, so it is not very consistent.

Maybe we should go back to doing `packet.(*dhcpv6.Message)`. Sorry for changing my mind here.

Reply:
I personally prefer more this way as we ensure that we are in the inner message 😅

I was tempted to modify also the code that is getting opt16/opt17 but I didn't wanted to break too many things

Reply:
Yeah, the issue now is that how we get the options is not consistent.